tests/experimental/pydantic/test_basic.py

```diff
@@ -271,6 +272,33 @@ class UserType:
     assert field3.type.of_type is str
 
 
+def test_type_with_fields_mutable_default():
+    empty_list = []
+
+    class User(pydantic.BaseModel):
+        groups: List[str]
+        friends: List[str] = empty_list
+
+    @strawberry.experimental.pydantic.type(User)
+    class UserType:
+        groups: strawberry.auto
+        friends: strawberry.auto
+
+    definition: TypeDefinition = UserType._type_definition
+    assert definition.name == "UserType"
+
+    [field1, field2] = definition.fields
+
+    assert field1.default == dataclasses.MISSING
+    assert field2.default == dataclasses.MISSING
+    assert field1.default_factory == dataclasses.MISSING
+    # check that we really made a copy
+    assert id(field2.default_factory()) != id(empty_list)
+    assert id(UserType(groups=["groups"]).friends) != id(empty_list)
```

Comment:
Why not use `is`/`is not`?

Reply:
wouldn't that be "less explicit"? anyone reading the test would need to understand that `is` refers to the reference rather than the hash. most of us would remember that, but for myself i forget the difference from time to time. i'll rather be explicit here so other maintainers can understand it at one glance

Reply:
`a is b` is exactly the same as `id(a) == id(b)`

Reply:
yes correct. i'm not opioniated on this so i'll change it to `is` :D

Reply:
The first 3 lines should also use `is` as `dataclasses.MISSING` is a singleton object instance:
```
    assert field1.default is dataclasses.MISSING
    assert field2.default is dataclasses.MISSING
    assert field1.default_factory is dataclasses.MISSING
```

Reply:
changed!